C/jets.c

```diff
@@ -405,6 +405,62 @@ RIGHTMOST_(64,8)
 RIGHTMOST_(64,16)
 RIGHTMOST_(64,32)
 
+#define LEFT_PAD_LOW_1_(bitsM)                                                 \
+/* left_pad_low_1_m : TWO |- TWO^m */                                          \
+bool left_pad_low_1_##bitsM(frameItem* dst, frameItem src, const txEnv* env) { \
+  (void) env; /* env is unused. */                                             \
+  bool bit = readBit(&src);                                                    \
+  write##bitsM(dst, bit);                                                      \
+  return true;                                                                 \
+}
+LEFT_PAD_LOW_1_(8)
+LEFT_PAD_LOW_1_(16)
+LEFT_PAD_LOW_1_(32)
+LEFT_PAD_LOW_1_(64)
+
+#define LEFT_PAD_LOW_(bitsN, bitsM)                                                    \
+/* left_pad_low_n_m : TWO^n |- TWO^m */                                                \
+bool left_pad_low_##bitsN##_##bitsM(frameItem* dst, frameItem src, const txEnv* env) { \
+  (void) env; /* env is unused. */                                                     \
+  for(int i = 0; i < (bitsM)/(bitsN) - 1; ++i) { write##bitsN(dst, 0); }               \
+  copyBits(dst, &src, (bitsN));                                                        \
+  return true;                                                                         \
+}
+LEFT_PAD_LOW_(8,16)
+LEFT_PAD_LOW_(8,32)
+LEFT_PAD_LOW_(8,64)
+LEFT_PAD_LOW_(16,32)
+LEFT_PAD_LOW_(16,64)
+LEFT_PAD_LOW_(32,64)
+
+#define RIGHT_PAD_LOW_1_(bitsM)                                                          \
+/* right_pad_low_1_m : TWO |- TWO^m */                                                   \
+bool right_pad_low_1_##bitsM(frameItem* dst, frameItem src, const txEnv* env) {          \
+  (void) env; /* env is unused. */                                                       \
+  bool bit = readBit(&src);                                                              \
+  write##bitsM(dst, (uint_fast##bitsM##_t)((uint_fast##bitsM##_t)bit << ((bitsM) - 1))); \
```

Comment:
a9f2c88d13783ee145c73d1545ee7a201edf01c9: `... << ((bitsM) - 1)`

If `bitsM` is zero, then we shift by negative amounts.

Reply:
Yes, but again we'd call `write0`.

By the way, I don't think you'll need to cast the result of the `<<` operation. (But it's not wrong either.)